Jenkinsfile

```diff
@@ -61,9 +61,38 @@ node("primary") {
     }
   }
 
-  if (isReleaseBranch()) {
+  if (isReleaseBranch() && isReleasable()) {
     stage("Publish") {
       sh 'yarn run publish'
     }
   }
 }
+
+def isNewerVersion(String current, String next) {
```

Comment:
Can these be added to `waluigi` instead? As you're going to need them for react and vue as well I believe?

Reply:
`isNewerVersion` for sure. `isRelesable` depends on having the specific scripts in the integration so not sure that was a great idea. Thoughts?

Reply:
if you update that to the following, then you shouldn't need the package.json scripts:
```
def isReleasable() {
  def packageName = sh(script: 'node -p -e "require(\'./package.json\').name"', returnStdout: true).trim()
  echo "Checking releasability for package: " + packageName
  if (packageName) {
    def remoteVersion = sh(script: "npm view ${packageName} version", returnStdout: true)
    echo "Validating remote version: " + remoteVersion
    def localVersion = sh(script: 'node -p -e "require(\'./package.json\').version"', returnStdout: true)
    echo "Validating local version: " + localVersion
    if (localVersion) {
      return isNewerVersion(remoteVersion, localVersion)
    }
  }
  return false
} 
```

Reply:
For reference, `node -p -e "require('./package.json').name"` and `node -p -e "require('./package.json').version"` were the changes I made, as that should achieve the same thing.

Reply:
I just read you can do `npm view . version` as well, which worked okay in my quick local testing.